src/Pdk/Plugin/Repository/PdkOrderRepository.php

```diff
@@ -91,6 +91,38 @@ public function get($input): PdkOrder
         });
     }
 
+    public function getByApiIdentifier(string $uuid): ?PdkOrder
+    {
+        /**
+         * Load the most recent orders to search for the apiIdentifier.
+         * Assuming change webhooks are fired for recent orders.
+         */
+        $orderIds = wc_get_orders([
```

Comment:
Op basis van de DB structuur zie ik idd geen betere manier om dit te doen. Het is serialized json dus kan ook geen JSON queries doen als we dat zouden willen.

Maar ik denk wel goed om dit een soort gepagineerd te doen, zodat het ook voor grote datasets werkt.
M.a.w. recursief elke keer 300 (of dan iets minder) results ophalen, als daar geen match in zit, dan een volgende set van 300 met een offset. Totdat er < 300 results in zitten en er dus niet meer resultaten over zijn, dan kun je de debug melding tonen en null returnen.

Reply:
Het gedrag is nu: als het te oud is, doen we geen update meer, dat was wel bewust (werkt in Magento ook zo). Maar ik vind het ook goed om de orders door te lopen tot een match is gevonden of er geen orders meer zijn. Ik zal het aanpassen.